libraries/provider_database_mysql_user.rb

```diff
@@ -84,6 +84,11 @@ def action_grant
         def exists?
           db.query("SELECT User,host from mysql.user WHERE User = '#{@new_resource.username}' AND host = '#{@new_resource.host}'").num_rows != 0
         end
+
+        def mysql_escape (s)
+          # having in mind that escaped string will be used within single quotes!
+          s.gsub(/'/,"''").gsub(/\\/,'\\\\\\')
```

Comment:
Why do you think that unsupported gem (added dependency) is better than I custom implemented quick-fix without added dependency? FYI: https://groups.google.com/forum/#!topic/rubyinstaller/nzpIDZqXO-Q/discussion